cot/src/lib.rs

```diff
@@ -172,3 +172,38 @@ pub type StatusCode = http::StatusCode;
 
 /// A type alias for an HTTP method.
 pub type Method = http::Method;
+
+/// A derive macro that automatically implements the [`FromRequestParts`] trait
+/// for structs.
+///
+/// This macro generates code to extract each field of the struct from HTTP
+/// request parts, making it easy to create composite extractors that combine
+/// multiple data sources from an incoming request.
+///
+/// The macro works by calling [`FromRequestParts::from_request_parts`] on each
+/// field's type, allowing you to compose extractors seamlessly. All fields must
+/// implement the [`FromRequestParts`] trait for the derivation to work.
+///
+/// # Requirements
+///
+/// - The target struct must have all fields implement [`FromRequestParts`]
+/// - Works with named fields, unnamed fields (tuple structs), and unit structs
+/// - The struct must be accessible where the macro is used
+///
+/// # Examples
+///
+/// ## Named Fields
+///
+/// ```no_run
+/// use cot::request::extractors::{Path, StaticFiles, UrlQuery};
+/// use cot::router::Urls;
+/// use cot_macros::FromRequestParts;
+/// use serde::Deserialize;
+///
+/// #[derive(Debug, FromRequestParts)]
+/// pub struct BaseContext {
+///     urls: Urls,
+///     static_files: StaticFiles,
+/// }
+/// ```
+pub use cot_macros::FromRequestParts;
```

Comment:
What if we move this to `src/request/extractors.rs`, effectively exporting `FromRequestParts` both as a trait and as a derive macro. Then, in `admin.rs` we wouldn't have to rename the `FromRequestParts` import into `FromRequestPartsTrait`, which would simplify things a little bit.